components/connector/cmd/main.go

```diff
@@ -49,13 +48,13 @@ func main() {
 
 	authContextMiddleware := authentication.NewAuthenticationContextMiddleware()
 
-	externalGqlServer, err := config.PrepareExternalGraphQLServer(cfg, certificateResolver, authContextMiddleware.PropagateAuthentication)
+	externalGqlServer, err := config.PrepareExternalGraphQLServer(cfg, certificateResolver, authContextMiddleware.PropagateAuthentication, correlation.AttachCorrelationIDToContext())
```

Comment:
Probably not, since calls to k8s components are not part of a request chain.

Reply:
It won't be enough. The context is not propagated - we basically use the `Get(name string, options metav1.GetOptions) (*v1.Secret, error)` method to get the resources. I just saw that in the newer version of the k8s go client we can also pass the context. I'll try to update it and see if everything works as expected.

On the other hand, the logs we can get from the K8s control plane are not that much - since GCP is managing the node where the CP is running, we cannot get the logs directly. We can get some auditlogs tho, but not for GET requests. But I'll definitely try to update the version.

Reply:
_Discussed offline:_
Since the calls that the Connector makes to the API Server are not initiated by a user request but are rather regular, there's no need for correlation IDs at this point.